lib/reader.ts

```diff
@@ -150,15 +160,15 @@ class ParquetReader {
    * and internal use cases. Consider using one of the open{File,Buffer} methods
    * instead
    */
-  constructor(metadata, envelopeReader, opts) {
+  constructor(metadata: NewFileMetaData, envelopeReader: ParquetEnvelopeReader, opts: BufferReaderOptions) {
     opts = opts || {};
     if (metadata.version != PARQUET_VERSION) {
       throw 'invalid parquet version';
     }
 
     // If metadata is a json file then we need to convert INT64 and CTIME
     if (metadata.json) {
-      const convert = (o) => {
+      const convert = (o: {[string: string]: any } ) => {
```

Comment:
One of the few times I used type `any`, o should be some time of object that has values of `parquetType` and `value` but function `convert` is used on line 196 with type NewFileMetaData

Reply:
@waylandli could it be type `unknown` instead?

Reply:
I tried but it's difficult based on the different ways its being used in that function. `o` has to be an object that can recursively be used due to convert calling itself with `o[key]` while also being an object with values that can be used to make a new Date/Buffer object in lines 175 and 177